app/styles/ui/_text-box.scss

```diff
@@ -27,5 +27,24 @@
   input {
     @include textboxish;
     @include textboxish-disabled;
+
+    // Customize the "clear" icon on search text inputs to match our
+    // theme icons and colors.
+    //
+    // To do so, we're creating an opaque 16x16 element with the background color
+    // that we want the icon to appear in and then apply the icon path
+    // as a mask, that way we can control the color dynamically based on
+    // our variables instead of hardcoding it in the SVG.
+    &[type='search']::-webkit-search-cancel-button {
+      -webkit-appearance: none;
+      width: 16px;
+      height: 16px;
+      margin-right: 0;
+      background-color: var(--text-color);
+      // The following SVG corresponds to the `x` octicon in 16px size:
+      // https://github.com/primer/octicons/blob/4661c1e0aa30c7d252318d2b003af782a6891089/icons/x-16.svg#L1
+      -webkit-mask-image: url('data:image/svg+xml;utf8,<svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" viewBox="0 0 16 16"><path fill-rule="evenodd" d="M3.72 3.72a.75.75 0 011.06 0L8 6.94l3.22-3.22a.75.75 0 111.06 1.06L9.06 8l3.22 3.22a.75.75 0 11-1.06 1.06L8 9.06l-3.22 3.22a.75.75 0 01-1.06-1.06L6.94 8 3.72 4.78a.75.75 0 010-1.06z"/></svg>');
```

Comment:
I've decided to use the current `master` version of the icon since it uses a square canvas (as opposed to the one corresponding to the version of the octicons that we're using, which has a canvas of 12x16 pixels) which makes it slightly easier to fit.

In terms of design, the only difference is that this newer icons is slightly thinner that the one in v9.1.1:

This icon:

<img width="408" alt="Screen Shot 2020-08-17 at 12 00 20" src="https://user-images.githubusercontent.com/408035/90384033-3f8f9980-e081-11ea-9bb4-93217fd4969c.png">

Icon in v9.1.1:

<img width="441" alt="Screen Shot 2020-08-17 at 12 01 01" src="https://user-images.githubusercontent.com/408035/90384106-56ce8700-e081-11ea-893b-b0357c1e5682.png">